src/mad_spark_alt/evolution/error_recovery.py

```
@@ -0,0 +1,372 @@
+                else:
+                    # Fallback for simple evaluators - return basic fitness
+                    return IndividualFitness(
+                        idea=idea,
+                        creativity_score=0.5,
+                        diversity_score=0.5,
+                        quality_score=0.5,
+                        overall_fitness=0.5,
+                    )
+
+            except asyncio.TimeoutError as e:
+                raise NetworkError(f"Evaluation timed out: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)